include/nlohmann/json.hpp

```diff
@@ -5643,6 +5670,7 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         // result.m_it.array_iterator = m_value.array->insert(pos.m_it.array_iterator, cnt, val);
         // but the return value of insert is missing in GCC 4.8, so it is written this way instead.
 
+        set_parents();
```

Comment:
Any particular reason not to follow the pattern from the other functions and only update the inserted object if the capacity didn't change?

Reply:
Oh, because this handles both single insert and range insert.

Reply:
Yes. I tried to add a special case, but did not figure a way to make it work.